### PR TITLE
Workflow configuration

### PR DESCRIPTION
**Pull request by noniagriconomie**
Hi,

Related to https://github.com/symfony/symfony-docs/issues/11043
I can also provide the 4.3 update (when 4.3 is feature freezed ? should we do 4.1/4.2 ?)

The review can be difficult as I reordered some parts, better read it on the SymfonyCloud build

Cheers

### CODE REVIEW

**Comment by noniagriconomie** [comment]
friendly ping @lyrixx as codeowner of workflow, reviews/comments appreciated :)

**Comment by javiereguiluz** [comment]
@noniagriconomie thanks for working on this! Sadly, we cannot reorder the options as proposed. We should keep the alphabetical sorting, which is what we use in all listings of config options across the docs.

**Comment by noniagriconomie** [comment]
@javiereguiluz ha ok, understood, i reordered them as "config priority"
I will revert no problem

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -2252,13 +2252,19 @@ Name of the workflow you want to create.
 audit_trail
 """""""""""
 
-**type**: ``array``
+**type**: ``bool``
```
Comment (noniagriconomie) [inline reference/configuration/framework.rst]: unsure of this one, why this one was array?

Comment (javiereguiluz) [reply reference/configuration/framework.rst]: Probably a copy+paste error.

Comment (lyrixx) [reply reference/configuration/framework.rst]: This is indeed a `bool`

**Review by lyrixx — APPROVED** [review]
:+1:

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -2252,13 +2252,19 @@ Name of the workflow you want to create.
...
+
+Whether to enable automatically the :class:`Symfony\\Component\\Workflow\\EventListener\\AuditTrailListener`.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
Whether to enable the :class:`Symfony\\Component\\Workflow\\EventListener\\AuditTrailListener` automatically.
```
reads more fluently

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -2276,11 +2282,15 @@ places
 
 **type**: ``array``
 
+All available places (**type**: ``string``) for this workflow configuration.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
All available places (**type**: ``string``) for the workflow configuration.
```

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -2276,11 +2282,15 @@ places
...
+
 supports
 """"""""
 
 **type**: ``string`` | ``array``
 
+Object(s) that is(are) supported by this workflow configuration.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
Object(s) that is(are) supported by the workflow configuration.
```

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -2293,11 +2303,13 @@ transitions
 
 Each marking store can define any of these options:
 
-* ``from`` (**type**: ``string``)
-* ``guard`` (**type**: ``string``) a :doc:`ExpressionLanguage </components/expression_language>`
+* ``from`` (**type**: ``string`` or ``array``) value from the ``places``,
+  multiple values are allowed for both ``workflow`` and ``state_machine``
```
Comment (OskarStark) [inline reference/configuration/framework.rst]: What about ending these description with a `;`?

Like:
```suggestion
  multiple values are allowed for both ``workflow`` and ``state_machine``;
```

**Comment by noniagriconomie** [comment]
@OskarStark I am currently in vacations
Feel free to commit your suggestions and advance the PR 👍🏻
Thanks

File: reference/configuration/framework.rst [hunk 6]
```diff
@@ -2293,11 +2303,13 @@ transitions
...
+* ``guard`` (**type**: ``string``) an :doc:`ExpressionLanguage </components/expression_language>`
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
* ``guard`` (**type**: ``string``) an :doc:`ExpressionLanguage </components/expression_language>`;
```

File: reference/configuration/framework.rst [hunk 7]
```diff
@@ -2293,11 +2303,13 @@ transitions
...
   compatible expression to block the transition
-* ``name`` (**type**: ``string``)
-* ``to`` (**type**: ``string``)
+* ``name`` (**type**: ``string``) the name of the transition
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
* ``name`` (**type**: ``string``) the name of the transition;
```

File: reference/configuration/framework.rst [hunk 8]
```diff
@@ -2293,11 +2303,13 @@ transitions
...
+* ``to`` (**type**: ``string`` or ``array``) value from the ``places``,
+  multiple values are allowed only for ``workflow``
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
  multiple values are allowed only for ``workflow``;
```

**Comment by OskarStark** [comment]
> Feel free to commit your suggestions and advance the PR 👍🏻

Enjoy your vacation 🌴 ☀️ , we can apply the changes when merging this PR!

👍 from my side @javiereguiluz

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Antoine! We merged this and made all changes proposed by @OskarStark while merging it.

**Pull request merged**